src/interchange/src/avro/schema.rs

```diff
@@ -136,11 +181,28 @@ fn get_named_columns<'a>(
     }
 }
 
+/// Get the single column corresponding to a schema node.
+/// It is an error if this node should correspond to more than one column
+/// (because it is an Essential Union in the sense described in the module docs).
 fn validate_schema_2(
     seen_avro_nodes: &mut HashSet<usize>,
     schema: SchemaNode,
 ) -> anyhow::Result<ScalarType> {
     Ok(match schema.inner {
+        SchemaPiece::Union(_) => {
+            let columns = get_union_columns(seen_avro_nodes, schema, None)?;
```

Comment:
`get_union_columns` should probably be renamed to `get_nonnull_union_columns` to make this more clear

Reply:
I named it _columns_, not _variants_, to make it clear that we're talking about the SQL-level concept, not the Avro-level one. This is the function that, given an Avro union, returns the series of all SQL columns it expands into.

Reply:
ah that makes sense!